woocommerce/woocommerce-android
# [Kotlin Warnings As Errors] Woo Commerce Module - Resolve Test Warnings & Enable All Warnings as Errors

Parent: #6816
Partially Closes: #6819



### Description


This PR resolves all `Test` related warnings for the `woocommerce` module:

- 139 x `coroutines`
    - 116 x `This declaration is experimental and its usage should be marked with '@kotlinx.coroutines.ExperimentalCoroutinesApi' or '@OptIn(kotlinx.coroutines.ExperimentalCoroutinesApi::class)'`
    - 10 x `'runBlockingTest(suspend TestCoroutineScope.() -> Unit): Unit' is deprecated. Use `runTest` instead to support completing from other dispatchers. Please see the migration guide for details: https://github.com/Kotlin/kotlinx.coroutines/blob/master/kotlinx-coroutines-test/MIGRATION.md`
    - 8 x `'advanceTimeBy(Long): Unit' is deprecated. The name of this function is misleading: it not only advances the time, but also runs the tasks scheduled *at* the ending moment.`
    - 2 x `'TestCoroutineDispatcher' is deprecated. The execution order of `TestCoroutineDispatcher` can be confusing, and the mechanism of pausing is typically misunderstood. Please use `StandardTestDispatcher` or `UnconfinedTestDispatcher` instead.`
    - 2 x `'TestCoroutineScope(CoroutineContext = ...): TestCoroutineScope' is deprecated. This constructs a `TestCoroutineScope` with a deprecated `CoroutineDispatcher` by default. Please use `createTestCoroutineScope` instead.`
    - 1 x `'runBlockingTest(CoroutineContext = ..., suspend TestCoroutineScope.() -> Unit): Unit' is deprecated. Use `runTest` instead to support completing from other dispatchers. Please see the migration guide for details: https://github.com/Kotlin/kotlinx.coroutines/blob/master/kotlinx-coroutines-test/MIGRATION.md`
- 1 x `deprecated`
    - 1 x `'XYX' is deprecated. Deprecated in Java`
- 5 x `cast`
    - 5 x `Unchecked cast: Xyz to Xyz`
- 15 x `other`
    - 1 x `Parameter 'xyz' is never used`
    - 9 x `Parameter 'xyz' is never used, could be renamed to _`
    - 5 x `Name contains characters which can cause problems on Windows: "`

Warnings Resolution List:

1. `coroutines`
    - Resolve experimental coroutine api warnings for tests. ([here](https://github.com/woocommerce/woocommerce-android/commit/8d476688c4ab76e5cd2179c50e64ad8c67a6edc4) and [here](https://github.com/woocommerce/woocommerce-android/pull/6902/commits/819670ac9accacf7ea10808a475a4228d1d5fde1))
    - [Resolve run blocking test warnings for card reader test.](https://github.com/woocommerce/woocommerce-android/commit/2c65ebf8c280a76c07ccd534dc2d6986e89605df)
    - [Resolve test coroutine scope warnings for currency form. test.](https://github.com/woocommerce/woocommerce-android/commit/1605351d7d9f4bf8849a853f19a735a35fac479b)
2. `deprecated`
    - [Resolve date deprecated warnings for test utils.](https://github.com/woocommerce/woocommerce-android/commit/46473fa61f451c8a8a3b333fb2e48998a23b56f2)
3. `other`
    - [Resolve parameter is never used warnings for test utils.](https://github.com/woocommerce/woocommerce-android/commit/326cf9cbe829b046344fd38bd7089315a6876cc8)
    - [Resolve parameter is never renamed to _ warnings for tests.](https://github.com/woocommerce/woocommerce-android/commit/def6ca71970dad287e9aedee53d18ae1f169b599)
    - [Resolve test function name characters test problems.](https://github.com/woocommerce/woocommerce-android/commit/5455d7cef25178a654d7f2c63e93f997c0d13f21)

Warnings Suppression List:

1. `coroutines`
    - [Suppress coroutine warnings for review moderation handler test.](https://github.com/woocommerce/woocommerce-android/commit/4a6304d4c8f3e6644bc3e42f0e88c99189a451bd)
2. `cast`
    - [Suppress unchecked cast warnings for tests.](https://github.com/woocommerce/woocommerce-android/commit/3910e4279d270e73b62cfde21be1a8a0521d1d07)

UI Test Related:

1. [Resolve or suppress various warnings for ui tests.](https://github.com/woocommerce/woocommerce-android/pull/6902/commits/933d14053dded03bfe052cd5eaa0891a3d83d6b4)

Refactor List:

1. [Move experimental coroutines api annotation to class level.](https://github.com/woocommerce/woocommerce-android/commit/af43338253c9842f7bcd67287a9fe37c4c6566ef)
2. [Refactor: Optimize imports for print shipping label view model test.](https://github.com/woocommerce/woocommerce-android/commit/1b5f6677b9517a8e3b7c909ecc7001cc75016a79)

In addition to the above warning resolution list, the below `Detekt` warnings got resolved as well, leaving Detekt's `baseline.xml` file 23 entries shorter:
1. [Resolve (no) wildcard imports detekt warns for mockito kotlin.](https://github.com/woocommerce/woocommerce-android/commit/c5e80449c37f72fc3fdf5dafd86b508bbf567225)
2. [Resolve (no) wildcard imports detekt warns for fluxc store.](https://github.com/woocommerce/woocommerce-android/commit/94bb067284037a5da7d04a282f074f44af84b8ed)
3. [Resolve (no) wildcard imports detekt warnings for kotlin test.](https://github.com/woocommerce/woocommerce-android/commit/cf4399c7f7b66b3845554161c82a43b24e7d0f69)
4. [Add back two wrongly removed detekt baseline entries.](https://github.com/woocommerce/woocommerce-android/commit/5622a736217e533618d70154823de9982327c352)

The `allWarningsAsErrors` configuration is currently applied on the module level in order to make sure that, as the overall [Kotlin Warnings as Errors](https://github.com/woocommerce/woocommerce-android/issues/6816) work is progressing, no new warnings are added to this module, which is already free of warnings.

Actually, with this [Enable all warnings as errors for woo commerce module.](https://github.com/woocommerce/woocommerce-android/commit/16cc45f170a9a0cda059d2d94dbbc1e0816334dc) commit in this PR, the [Kotlin Warnings as Errors](https://github.com/woocommerce/woocommerce-android/issues/6816) work is complete on all modules.

When this PR gets merged, this module level `allWarningsAsErrors` configuration will be replaced by a root level `allWarningsAsErrors` configuration that will be applied by default to all modules (see [here](https://github.com/woocommerce/woocommerce-android/issues/6820)). 

-----

PS: @nbradbury I added you as the main reviewer, that is, in addition to the @woocommerce/owl-team team itself, but randomly, since I just wanted someone from the Woo Android team to sing-off on that change for WCAndroid as well. PS: I am going to randomly add more of you in those PRs just so you become more aware of this change and how close we are on enabling `allWarningsAsErrors` by default everywhere. 🥇

### Testing instructions


- Verify that all the CI checks are successful.
- PS: There is no need to smoke test the app and verify that everything works as expected as this PR is mostly test related (give or take the `OrderDetailRepository` related imports that got updated).

### Images/gif
